openmdao/solvers/linesearch/backtracking.py

```diff
@@ -191,7 +196,16 @@ def _iter_initialize(self):
         elif self.options['bound_enforcement'] == 'wall':
             u._enforce_bounds_wall(du, self.alpha, system._lower_bounds, system._upper_bounds)
 
-        norm = self._iter_get_norm()
+        try:
+            norm = self._iter_get_norm()
+
+        except AnalysisError as err:
+            if self.options['retry_on_analysis_error']:
+                self._analysis_error_raised = True
+            else:
+                raise AnalysisError(str(err))
```

Comment:
If we re-raise the exception in this way, we'll lose the traceback information from the original exception.  Instead we should do something like we did in clippy:

from six import reraise

  .. in your except block...
    exc = sys.exc_info()
    reraise(exc[0], exc[1], exc[2])


If you want to replace the exception with a new one but keep the original traceback, you can replace
exc[1] with the new exception object.